Neos.Flow/Classes/Http/Middleware/SecurityEntryPointMiddleware.php

```diff
@@ -0,0 +1,86 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Http\Middleware;
+
+use GuzzleHttp\Psr7\Response;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Log\Utility\LogEnvironment;
+use Neos\Flow\Security\Authentication\Token\SessionlessTokenInterface;
+use Neos\Flow\Security\Authentication\TokenInterface;
+use Neos\Flow\Security\Context;
+use Neos\Flow\Security\Exception\AuthenticationRequiredException;
+use Psr\Log\LoggerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * A HTTP middleware that handles authentication exceptions that were thrown by the dispatcher (@see \Neos\Flow\Mvc\Dispatcher::dispatch()) and
+ * * rethrows the exception if no token with Entry Point is authenticated
+ * * or otherwise invokes the Entry Point of all authenticated tokens
+ */
+class SecurityEntryPointMiddleware implements MiddlewareInterface
+{
+    /**
+     * @Flow\Inject
+     * @var Context
+     */
+    protected $securityContext;
+
+    /**
+     * @Flow\Inject(name="Neos.Flow:SecurityLogger")
+     * @var LoggerInterface
+     */
+    protected $securityLogger;
+
+    /**
+     * @inheritDoc
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $next): ResponseInterface
+    {
+        try {
+            return $next->handle($request);
+        } catch (AuthenticationRequiredException $authenticationException) {
+            /** @var TokenInterface[] $tokensWithEntryPoint */
+            $tokensWithEntryPoint = array_filter($this->securityContext->getAuthenticationTokens(), static function (TokenInterface $token) {
+                return $token->getAuthenticationEntryPoint() !== null;
+            });
+
+            if ($tokensWithEntryPoint === []) {
+                $this->securityLogger->notice('No authentication entry point found for active tokens, therefore cannot authenticate or redirect to authentication automatically.');
+                throw $authenticationException;
+            }
+
+            $response = $this->buildHttpResponse();
+            foreach ($tokensWithEntryPoint as $token) {
+                if ($token->isAuthenticated()) {
```

Comment:
This check wasn't there before. I'm not 100% sure about this, but shouldn't tokens that are authenticated **not** lead to a authentication entry point invokation?

Reply:
that double negation was too much for my monday-night braing :)

Reply:
Ah, I forgot to mention this one (three) line(s) in our call @bwaidelich - maybe you have an idea if I am missing something or this causes bad side-effects if we add them?

Reply:
Sure, let's talk about it tomorrow!

Reply:
Just for the record: We decided to remove this again and look at it separately - see #2234